# Baseline Training

Train a simple baseline model, report metrics, and save model plus metrics artifacts. This tests the normal agent loop for a first useful model.

In [1]:
from pathlib import Path

exec(Path("samples/ml_tasks/baseline_training.py").read_text(encoding="utf-8"))


rows: 900
train_rows: 648
test_rows: 252
model_artifact: artifacts/ml_tasks/baseline_model.json
metrics_artifact: artifacts/ml_tasks/baseline_metrics.json
epoch   1: loss=0.6931
epoch  23: loss=0.6256
epoch  46: loss=0.5823
epoch  69: loss=0.5548
epoch  92: loss=0.5365
epoch 115: loss=0.5237
epoch 138: loss=0.5146
epoch 140: loss=0.5139
test_metrics:
- tp: 108
- fp: 27
- tn: 89
- fn: 28
- precision: 0.8000
- recall: 0.7941
- accuracy: 0.7817
- f1: 0.7970
- brier: 0.1646


tp,108
fp,27
tn,89
fn,28
precision,0.8000
recall,0.7941
accuracy,0.7817
f1,0.7970
brier,0.1646


## Finding

The clean run produced a usable baseline-training story: the notebook trains on 648 rows, evaluates on 252 test rows, and saves both the model and metrics artifacts.

Key test metrics: accuracy 0.7817, precision 0.8000, recall 0.7941, F1 0.7970, Brier 0.1646. Artifacts: `artifacts/ml_tasks/baseline_model.json` and `artifacts/ml_tasks/baseline_metrics.json`.
